### ISSUE TITLE
Most tests fail if @visx/scale is a dependency in package.json and Test alias does not alias d3-scale properly

### ISSUE THREAD

**Issue by AddictArts** [vitest-dev/vitest#3539]
### Describe the bug

Currently don't know where the issue is. I'll start it as a bug report, but it may be a documentation issue and how to address this case.

Note: None of the tests use anything `@visx` or `d3-array`. The tests are just in the same package that has react components that use it. Somehow vitest thinks it needs to check those files.

Using the test.alias function for `@visx/scale` to load an esm runs into issues with `d3-scale`. The below works great until you need d3-scale.

```
  11   │   test: {
  12   │     alias: [
  13   │       { find: /^@visx\/scale$/, replacement: '@visx/scale/esm/index.js' },
  14   │     ],
```

Once that happens you will get the error, module `require()` error. It shows a path of `@visx/scale/lib/scale`; however, that alias should be putting everything requested in `@visx/scale/esm` I would think. I tried replacement of `@visx/cales/esm` instead of directly to the `index.js`. I don't think `resolve.mainFields` would help either as it points to `esm/index.js` which is exactly what the alias does.

The error points to a require of `node_modules/d3-scale/src/index.js`, so I tired `deps.inline: 'd3-scale'`, many variations on that and that did not work.

There is one issue and one discussion that seem like the point this out as being not supported or available yet.

https://github.com/vitest-dev/vitest/issues/2740 and https://github.com/vitest-dev/vitest/discussions/3221

### Reproduction

Use `@visx/scale` in a project, example just copy and paste the **Brush** example from `@visx` gallery into a project.

https://airbnb.io/visx/brush

Any vitest tests in the same project even if they are not accessing that react component or using any `@visx` things, import or other, will still result in the error.

### System Info

```shell
Windows 10
Vite latest
Vitest latest
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
Vitest doesn't import anything you didn't import. If the package is used, it means it was imported somewhere.

**Comment by sheremet-va** [comment]
> Not true from what I can see when `includeSource` and using inline tests

Vitest imports every file that matches the glob pattern and has `import.meta.vitest` inside. If `@visx/scale` is evaluated, it means it's imported somewhere in the import chain of that file.

**Comment by AddictArts** [comment]
> Your link in reproduction doesn't reference Vitest. Please, provide a runnable reproduction.

I was working on it, but then as a last ditch effort I removed the test alias and just put this:

```
  11 + │   resolve: {
  12 + │     mainFields: ['module'],
  13 + │   },
```

Now the in source tests work and other react ones. It is worth noting I checked all the code `ripgrep` for d3-scale and we were not importing it. My suspicion is that the alias was not effective for the entire `@visx/scale` module and that resulted in a path to a call to require().
